sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/nullFunctions.scala

```diff
@@ -210,14 +210,58 @@ case class IsNotNull(child: Expression) extends UnaryExpression with Predicate {
   }
 }
 
+/**
+ * A predicate that is evaluated to be true if there are at least `n` null values.
+ */
+case class AtLeastNNulls(n: Int, children: Seq[Expression]) extends Predicate {
```

Comment:
Yeah. Because `null` means `Unknown`, so when you have a predicate `null = null`, the result is false (meaning `Unknown`). But for `NaN`, in our current semantic, two `NaN` are equal.